src/api/prop.js

```diff
@@ -26,9 +26,10 @@ export const boolean = create({
 });
 
 export const number = create({
-  coerce: alwaysUndefinedIfEmptyOrNumber,
-  deserialize: alwaysUndefinedIfEmptyOrNumber,
-  serialize: alwaysUndefinedIfEmptyOrNumber
+  default: 0,
+  coerce: alwaysZeroIfNotANumberOrNumber,
```

Comment:
That works in every case except where the input value is a string e.g. `'test'`. In that case it would get coerced to `NaN` rather than `0` because `Number('test') == NaN`.